.changeset/wise-cameras-agree.md

```diff
@@ -0,0 +1,22 @@
+---
+'@astrojs/cloudflare': major
+---
+
+The configuration `build.split` and `build.excludeMiddleware` are deprecated.
+
+Configuration that were inside the astro configuration, are now moved inside the adapter:
+
+```diff
+import {defineConfig} from "astro/config";
+import cloudflare from '@astrojs/cloudflare';
+
+export default defineConfig({
+-    build: {
+-        split: true
+-    },
+-    adapter: cloudflare()
++    adapter: cloudflare({
++        functionPerRoute: true
```

Comment:
```suggestion
+    adapter: cloudflare({
+        mode: 'directory',
+        functionPerRoute: true
```
I'm not sure whether this is the case? But the way it was described below made it sound like this option was only for when 'directory' mode was configured. If so, I'd strongly suggest showing that here.

(If it's not, the README needs more work because it only discusses this option in terms of "in directory mode", so that's what I've been going off of.)